metals/src/main/scala/scala/meta/internal/metals/codeactions/CustomCodeActionKind.scala

```diff
@@ -0,0 +1,5 @@
+package scala.meta.internal.metals.codeactions
+
+object CustomCodeActionKind {
+  final val SourceAddMissingImports = "source.addMissingImports"
```

Comment:
Why a custom kind? Don't we have to implement it in each editor now? I would use the normal source action kind

Reply:
You should be able to set it to autoSave if needed even if it's a source action

Reply:
> You should be able to set it to autoSave if needed even if it's a source action

@tgodzik can you explain how you would do this? I'm looking through VS code docs, but they seem to suggest we should have different kinds (starting with `source`). Example at the to top of <https://code.visualstudio.com/docs/editing/refactoring#_code-actions-on-save> has a number of source actions.

The config I want is something like

```
"editor.codeActionsOnSave": {
    "source.addMissingImports": "always",
},
```

How can that be achieved with what was merged?

Reply:
I would have assumed that the config above would be what is needed. That doesn't work? :O

Reply:
Maybe I'm missing something, but where is "source.addMissingImports" in the final version of the PR? I thought that would be the CodeActionKind which got switched back to just "source"... 🤔

Reply:
`"source": "always"` would do it now I guess, but that will run everything :/ 

We could instead do a custom source action only if it's supported by a client, which would mean having a possibility to declare a custom one.

Reply:
My read of the LSP spec is that this should be fine - the actions are expected to be hierarchical in nature (so starting with source in our case), the server and the client can declare what they support in capabilities, but then the client should be robust to receiving other actions too (and crucially I don't think there's any editor-specific things metals should need to do). On phone, so might be misreading. I'll take a closer look tonight and then might amend my PR to add back the action.

Reply:
When I used this custom action it worked natively in vscode without any specific changed. 
As soon as we declare it the IDE accepts it

Reply:
ach, so it might have been my bad and it works without an issue. We can return to using the custom action in that case if anyone is willing to do a PR?

Reply:
If it can wait for 6 or so hours, I'll tack it onto my PR for adding a source action to remove bad imports. Otherwise happy to wait and rebase

Reply:
Done in https://github.com/scalameta/metals/pull/7416 now.